src/slice.cr

```diff
@@ -239,6 +239,68 @@ struct Slice(T)
     pointer(count).copy_to(target, count)
   end
 
+  # Copies the contents of this slice into *target*.
+  #
+  # Truncates if this slice doesn't fit into the destination slice.
```

Comment:
The point of the API is that it copies the size of the smaller slice, and returns the amount of bytes it copies. I think this API is sane, but i'm not sure, i'm willing to be convinced either way.